pkg/iter/entry_iterator.go

```diff
@@ -320,25 +318,56 @@ func NewSortEntryIterator(is []EntryIterator, direction logproto.Direction) Entr
 	result := &entrySortIterator{is: is}
 	switch direction {
 	case logproto.BACKWARD:
-		result.heap = &iteratorSortHeap{iteratorHeap: make([]EntryIterator, 0, len(is)), byAscendingTime: false, byAlphabetical: true}
+		result.byAscendingTime = false
+		result.byAlphabetical = true
```

Comment:
Is there any time we don't want `byAlphabetical` to be true? Could this just go in the initialization?

Reply:
The `main` version has the same logic. @cyriltovena when would `byAlphabetical` be false?

Reply:
The heap was shared with the Merge iterator which is not anymore so I think we can get rid of it.

The merge one is also always going to be non alphabetical I believe.

Reply:
Done.